### ISSUE TITLE
C0325: Unnecessary parens after 'not' keyword (superfluous-parens) false positive with walrus operator

### ISSUE THREAD

**Issue by mfrancis95** [PyCQA/pylint#3788]
### Steps to reproduce
```
if not (a := 5):
    print(a)
```

### Current behavior
`C0325: Unnecessary parens after 'not' keyword (superfluous-parens)`.

### Expected behavior
No warning.

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0]
```

**Comment by AWhetter** [comment]
I can reproduce this on master. As far as I'm aware, we don't have very good support for walrus operators yet.

**Comment by conqp** [comment]
Possible duplicate of https://github.com/PyCQA/pylint/issues/3249

**Comment by mfrancis95** [comment]
@conqp yes.

**Comment by PCManticore** [comment]
Thanks for the report @mfrancis95 Closing as a duplicate of #3249
